src/legacy/server/saved_objects/service/lib/repository.js

```diff
@@ -369,7 +380,7 @@ export class SavedObjectsRepository {
     }
 
     const esOptions = {
-      index: this._index,
+      index: this._getIndexForType(type),
```

Comment:
`type` could be a `string | string[]` but you're only handling the `string` case.